xbmc/filesystem/ShoutcastFile.h

```diff
@@ -48,6 +48,10 @@ class CShoutcastFile : public IFile, public CThread
   int m_discarded; // data used for tags
   int m_currint;
   char* m_buffer; // buffer used for tags
+  bool haveExtraData = false;
+  // use a dummy value here so that streams that supply no data will still have the station name
+  // set correctly.
+  std::string oldTitle = "dummy data";
```

Comment:
this needs to be translatable to other languages. use a resource string here, not a hard coded english value.

Reply:
Why, it's never displayed?  Its only there to ensure that the first time the title is compared to the oldTitle it will return true.  Otherwise, if its just empty and the station sends empty (or no) data the test will be false and the station name will not be set.

Reply:
This is what I call a hack.